spec/lib/utils/pdf_to_text_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+RSpec.describe DiscourseAi::Utils::PdfToText do
+  fab!(:llm_model)
+  fab!(:user)
+  let(:pdf) { plugin_file_from_fixtures("2-page.pdf", "rag") }
+  let(:upload) { UploadCreator.new(pdf, "2-page.pdf").create_for(Discourse.system_user.id) }
+  before { SiteSetting.authorized_extensions = "pdf|png|jpg|jpeg" }
+
+  describe "#extract_text" do
+    it "extracts text from PDF pages" do
+      pdf_to_text = described_class.new(upload: upload)
+      pages = []
+      pdf_to_text.extract_text { |page| pages << page }
+
+      expect(pages).to eq(["Page 1", "Page 2"])
+    end
+  end
+
+  context "when improving PDF extraction with LLM" do
+    it "can properly simulate a file" do
+      if ENV["CI"]
+        skip "This test requires imagemagick is installed with ghostscript support - which is not available in CI"
```

Comment:
I'm alittle confused here because CI runs tests with the `discourse/discourse_test:release` image that contains the same imagemagick binary as production. If this feature doesn't work in CI, it would mean that it won't work in production as well.

Reply:
I tried and it was missing gs... which is odd, I guess we can double check here, was sure @xfalcox added it.

Reply:
@tgxworld looks like this regressed on https://github.com/discourse/discourse_docker/pull/889. Can you help us there?